### PR TITLE
build fixes

### PR DESCRIPTION
**Pull request by vtjnash**
a couple minor fixes I had lying around

this'll force everyone to rebuild llvm, to make sure you don't have one of the corrupt builds (which you have if you've been doing incremental builds over the last couple of days)

### CODE REVIEW

File: deps/llvm.mk [hunk 1]
```diff
@@ -336,6 +336,7 @@ endif
 ifneq ($(LLVM_LLDB_TAR),)
 	$(JLCHECKSUM) $(LLVM_LLDB_TAR)
 endif
+	-rm -rf $(LLVM_SRC_DIR)
```
Comment (tkelman) [inline deps/llvm.mk]: kinda drastic

File: src/signal-handling.c [hunk 2]
```diff
@@ -53,13 +53,6 @@ JL_DLLEXPORT void jl_exit_on_sigint(int on)
     exit_on_sigint = on;
 }
 
-// what to do on SIGINT
-JL_DLLEXPORT void jl_sigint_action(void)
```
Comment (yuyichao) [inline src/signal-handling.c]: I left this in since this is in `julia.h`, hopefully no one is using this...

Comment (vtjnash) [inline src/signal-handling.c]: we eventually need to go through and remove all of the non-exported API stuff from there. I just keep doing a (very) little at a time (like here)

File: deps/llvm.mk (hunk 1 above)
Comment (vtjnash) [inline deps/llvm.mk]: but entirely correct

File: src/signal-handling.c (hunk 2 above)
Comment (tkelman) [inline src/signal-handling.c]: I don't see any registered Julia packages ccall'ing into it. Who knows about unregistered or embedding code though.

File: deps/llvm.mk (hunk 1 above)
Comment (tkelman) [inline deps/llvm.mk]: for the people who did rebuild from scratch recently, or haven't rebuilt since my commits, would touching cmakelists be equivalent?

Comment (vtjnash) [inline deps/llvm.mk]: Someone could touch this build target manually or we could revert to testing for configure. But from the time stamp on cmakelist, we can't tell if the user has a good or corrupted build.

Comment (tkelman) [inline deps/llvm.mk]: mind saving everyone time and announce on the mailing list and relevant issues that people should do that if they've already rebuilt in the last few days and have a good build?

Comment (tkelman) [inline deps/llvm.mk]: including on travis where merging this would delete the cached llvm and occupy travis for several hours

Comment (tkelman) [inline deps/llvm.mk]: do the queue a favor and don't merge anything until https://travis-ci.org/JuliaLang/julia/builds/148541438 is done

**Pull request merged**